iceoryx_hoofs/include/iceoryx_hoofs/error_reporting/error_forwarding.hpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_ERROR_REPORTING_ERROR_FORWARDING_HPP
+#define IOX_HOOFS_ERROR_REPORTING_ERROR_FORWARDING_HPP
+
+#include "iceoryx_hoofs/error_reporting/error_kind.hpp"
+#include "iceoryx_hoofs/error_reporting/source_location.hpp"
+
+// to establish connection to the custom implementation
+#include "iceoryx_hoofs/error_reporting/custom/error_reporting.hpp"
```

Comment:
Yes. This is not the best solution and we can improve this somehow with Cmake magic I think (but did not consider what the best mechanism would be). The crucial point is that the header has to be selected at build definition time or compile time at the latest.

The  non-custom code part knows it has to look into `iceoryx_hoofs/error_reporting/custom/error_reporting.hpp` for the definition (a compile time contract). Failing this, it will not work.

For now it is good enough. Can be improved once we consider more customization.